twitter/searcher.py

```diff
@@ -168,19 +178,70 @@ def get_media_url_from_variants(variants: List[Dict[str, str]]):
 
 # TODO: hook up to s3 bucket
 def log_last_processed_id(last_processed_id: int, last_processed_time_stamp: str):
-    msg = {"last_processed_id": last_processed_id, "last_processed_time_stamp": last_processed_time_stamp}
+    msg = {
+        "last_processed_id": last_processed_id,
+        "last_processed_time_stamp": last_processed_time_stamp,
+    }
     with open(TWITTER_LAST_RUN_FILENAME, "w") as f:
         f.write(json.dumps(msg))
-    # s3_client = boto_client("s3")
-    # response = s3_client.upload_file(TWITTER_LAST_RUN_FILENAME, TWITTER_LAST_RUN_BUCKET, TWITTER_LAST_RUN_FILENAME)
-    # return response
 
 
-def get_since_id_from_file():
+def get_since_id_from_file(self):
     with open(TWITTER_LAST_RUN_FILENAME, "r") as f:
         d = json.load(f)
     return d.get("last_processed_id")
 
 
+class LocalCache:
+    def log_last_processed_id(
+        self, last_processed_id: int, last_processed_time_stamp: str
+    ):
+        msg = {
+            "last_processed_id": last_processed_id,
+            "last_processed_time_stamp": last_processed_time_stamp,
+        }
+        with open(TWITTER_LAST_RUN_FILENAME, "w") as f:
+            f.write(json.dumps(msg))
+
+    def get_since_id_from_file(self):
+        with open(TWITTER_LAST_RUN_FILENAME, "r") as f:
+            d = json.load(f)
+        return d.get("last_processed_id")
+
+
+class S3Cache:
+    def __init__(self):
+        self.file = None
+        self.client = boto3.client(
+            "s3",
+            # Hard coded strings as credentials, not recommended.
+            aws_access_key_id=os.getenv("AWS_ACCESS_KEY"),
+            aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS"),
+        )
+        self.local_cache = LocalCache()
+
+    def log_last_processed_id(
+        self, last_processed_id: int, last_processed_time_stamp: str
+    ):
+        self.local_cache.log_last_processed_id(
+            last_processed_id, last_processed_time_stamp
+        )
+
+        response = self.client.upload_file(
+            TWITTER_LAST_RUN_FILENAME,
+            TWITTER_LAST_RUN_BUCKET,
+            TWITTER_LAST_RUN_FILENAME,
+        )
+        return response
+
+    def get_since_id_from_file(self):
+        self.client.download_file(
+            TWITTER_LAST_RUN_BUCKET,
```

Comment:
Just curious, why is the order of args different between upload and download

Reply:
Aws craziness 

download_file(Bucket, Key, Filename, ExtraArgs=None, Callback=None, Config=None)¶
upload_file(Filename, Bucket, Key, ExtraArgs=None, Callback=None, Config=None)